### PR TITLE
Switch to aiohttp masking method

### PR DESCRIPTION
**Pull request by pgjones**
This utilises the masking method used in aiohttp as it has better
performance, in addition the aiohttp test is also used. With thanks to
@willmcgugan for the original implementation.

This also makes it more explicit that wsaccel is an optional extra, by
adding it as an extra requirement in the setup.py.

### CODE REVIEW

File: wsproto/frame_protocol.py [hunk 1]
```diff
@@ -19,14 +18,21 @@
...
-            maskbytes = self._maskbytes
-            return bytearray(b ^ next(maskbytes) for b in bytearray(data))
+            if data:
+                data_array = bytearray(data)
+                a, b, c, d = (_XOR_TABLE[n] for n in self._masking_key)
+                data_array[::4] = data_array[::4].translate(a)
+                data_array[1::4] = data_array[1::4].translate(b)
+                data_array[2::4] = data_array[2::4].translate(c)
+                data_array[3::4] = data_array[3::4].translate(d)
+            return bytes(data_array)
```
Comment (willmcgugan) [inline wsproto/frame_protocol.py]: if `data` is empty, then `data_array` wouldn't exist at this point.

**Comment by njsmith** [comment]
> This also makes it more explicit that wsaccel is an optional extra, by adding it as an extra requirement in the setup.py.

Why do we still support wsaccel at all, if the pure-Python version is dramatically faster?

**Comment by pgjones** [comment]
> if the pure-Python version is dramatically faster

From https://www.willmcgugan.com/blog/tech/post/speeding-up-websockets-60x/

I'm not sure it is dramatically faster, I think the benchmark shows it to be a little slower (orange versus green). That said it is close enough that I'll update this to drop wsaccel support - I suspect it is rarely used by wsproto users anyway (given it was only mentioned in the code).

**Comment by njsmith** [comment]
Ah right, I forgot – the pure python versions are about as fast as wsaccel, and the newer accelerator modules are dramatically faster.

File: wsproto/frame_protocol.py (hunk 1 above)
Comment (pgjones) [reply wsproto/frame_protocol.py]: Thanks

**Comment by pgjones** [comment]
I've fixed the build test (see the key rotation), I'll merge tomorrow unless anyone disagrees.

**Pull request merged**